src/compressed_tensors/quantization/lifecycle/forward.py

```diff
@@ -199,24 +201,40 @@ def _process_quantization(
             sc = scale[:, i].view(-1, 1)
             zp = zero_point[:, i].view(-1, 1) if zero_point is not None else None
 
-            idx = i * group_size
-            if do_quantize:
-                output[:, idx : (idx + group_size)] = _quantize(
-                    x[:, idx : (idx + group_size)],
-                    sc,
-                    zp,
-                    q_min,
-                    q_max,
-                    args,
-                    dtype=dtype,
-                )
-            if do_dequantize:
-                input = (
-                    output[:, idx : (idx + group_size)]
-                    if do_quantize
-                    else x[:, idx : (idx + group_size)]
-                )
-                output[:, idx : (idx + group_size)] = _dequantize(input, sc, zp)
+            if g_idx is not None and i in g_idx:
```

Comment:
let's add a faster check if g_idx is initialized (ie outside of loop just run `g_idx != -1`)

Reply:
add in-line comment explaining - also update docstring for g_idx where appropriate